src/App.tsx

```diff
@@ -1,14 +1,74 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, {
+  useEffect,
+  useState,
+  useCallback,
+  useMemo,
+} from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [status, setSelectedStatus] = useState('all');
+  const [searchQuery, setSearchQuery] = useState('');
+  const [selectedTodoId, setSelectedTodoId] = useState(0);
+
+  useEffect(() => {
+    getTodos().then(todo => setTodos(todo));
+  }, []);
+
+  const visibleTodos = useMemo(() => todos.filter(todo => {
+    const isQueryMatch = todo.title.toLowerCase().includes(searchQuery.toLowerCase().trim());
+    let selectedFilterStatus = true;
+
+    switch (status) {
+      case 'active':
+        selectedFilterStatus = !todo.completed;
+        break;
+
+      case 'completed':
+        selectedFilterStatus = todo.completed;
+        break;
+
+      default:
+        selectedFilterStatus = true;
+    }
+
+    return isQueryMatch && selectedFilterStatus;
+  }), [todos, status, searchQuery]);
```

Comment:
In the future, big parts of code are better moved into separate files. It's extremely improved code readability